libevm/VM.cpp

```diff
@@ -643,7 +657,296 @@ void VM::interpretCases()
 					number &= mask;
 			}
 		}
-		NEXT
+		NEXT		
+
+#if EIP_616
+		
+		CASE(XADD)
+		{
+			ON_OP();
+			updateIOGas();
+
+			uint8_t nt = m_code[++m_PC]; 
+			xadd(nt);
+			++m_PC;
+		}
+		CONTINUE
+	         
+		CASE(XMUL)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xmul(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XSUB)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xsub(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XDIV)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xdiv(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XSDIV)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xsdiv(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+        
+		CASE(XMOD)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xmod(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XSMOD)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xsmod(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+        
+		CASE(XLT)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xlt(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+          
+		CASE(XGT)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xgt(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+          
+		CASE(XSLT)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xslt(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XSGT)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xsgt(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XEQ)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xeq(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+          
+		CASE(XISZERO)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xzero(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+      
+		CASE(XAND)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xand(m_code[++m_PC]); ++m_PC;
+		}
+		CONTINUE
+         
+		CASE(XXOR)
+		{
+			ON_OP();
+			updateIOGas();
+
+			xoor(m_code[++m_PC]); ++m_PC;
```

Comment:
Wrong.  Seems `xor` is now a keyword, so I called the function `xoor`, and it needs a case to call it too.